Fixed several links + improved the linkchecker (#2371)
segmentio/segment-docs: src/connections/destinations/catalog/castle/index.md

)
```

> info ""
> Segment requires either `user_id` or `anonymous_id` for the request to be processed. If you don't know which user generated the failed login create a UUID and provide it as `anonymous_id`

## Extending server-side tracking with request properties
Tracking events from your server-side is crucial to prevent requests from getting blocked by malicious actors. This is recommended for all [Castle's reserved events](https://docs.castle.io/docs/events), such as logins and password changes.

> warning "" 

> > Server-side `track` events are dropped by Castle unless they contain the properties listed below. `identify` calls still create or update a user, but do not create a device if these properties are missing:

Reviewer: ```suggestion
> Server-side `track` events are dropped by Castle unless they contain the properties listed below. `identify` calls still create or update a user, but don't create a device if these properties are missing:
```

## Requesting a risk score
Castle's adaptive authentication tells you whether to allow access, initiate a second factor of authentication, or log out the user.

> Since all Segment calls are called asynchronously, you will need to use Castle's native SDKs to perform [adaptive authentication](https://docs.castle.io/docs/authentication-method).

Reviewer: ```suggestion
Since all Segment calls are called asynchronously, you'll need to use Castle's native SDKs to perform [adaptive authentication](https://docs.castle.io/docs/authentication-method).
```
